src/System.CommandLine/Parsing/StringExtensions.cs

```diff
@@ -566,7 +566,8 @@ IEnumerable<string> SplitLine(string line)
 
             for (var commandAliasIndex = 0; commandAliasIndex < command.RawAliases.Count; commandAliasIndex++)
             {
-                var commandAlias = command.RawAliases[commandAliasIndex];
+                var commandAlias = command.RawAliases.ToArray()[commandAliasIndex];
```

Comment:
Any reason to not use ElementAt here?
```suggestion
                var commandAlias = command.RawAliases.ElementAt(commandAliasIndex);
```

Reply:
None at all. A coupe of these were stabs at moving to `for` loops that dead ended at the `HashSet`.